extmod/ssl/ssl.py

```diff
@@ -0,0 +1,172 @@
+from ussl import *
+import ussl as _ussl
+
+# constants
+PROTOCOL_TLS_CLIENT = const(0)
+PROTOCOL_TLS_SERVER = const(1)
+
+
+# backwards compatibility even after C code is deprecated
+def wrap_socket(
+    sock,
+    server_side=False,
+    key=None,
+    cert=None,
+    cert_reqs=CERT_NONE,
+    cadata=None,
+    server_hostname=None,
+    do_handshake=True,
+    keyfile=None,
+    certfile=None,
+    ciphers="",
+):
+    ctx = _ussl.ctx_init()
+    if keyfile:
+        key = keyfile
+    if certfile:
+        cert = certfile
+    if key is not None:  # and (cert is not None):
+        ctx.load_certchain(key=key, cert=cert)
+    if cadata:
+        ctx.load_cadata(cadata)
+    if ciphers:
+        ctx.set_ciphers(ciphers.split(":"))
+    return ctx.wrap_socket(
+        sock,
+        server_side=server_side,
+        cert_reqs=cert_reqs,
+        server_hostname=server_hostname,
+        do_handshake=do_handshake,
+    )
+
+
+class SSLContext:
```

Comment:
I think this should inherit from the native "ssl context" class, and add methods that are missing, rather than have a `ctx` member.